campaignion_foundation.js

```diff
@@ -156,4 +156,35 @@ Drupal.behaviors.payment_slide.attach = function (context, settings) {
   });
 };
 
+/**
+ * Provide an event to act on form step changes.
+ *
+ * If no form step is yet known, assume we are initially loading a page with a
+ * webform.
+ *
+ * You can use this event to conditionally show/hide parts of a page.
+ *
+ * NB: We currently need to use native events to allow dispatching between
+ * different versions of jQuery (`.trigger()` and `.on()` communication
+ * between the Drupal's `$` and the the mo-foundation-base's `$` does not
+ * work).
+ */
+Drupal.behaviors.form_step_data = {};
+Drupal.behaviors.form_step_data.attach = function (context, settings) {
+  // current step is always a number > 0 if the settings information is correct
+  let currentStep = settings.campaignion_foundation.webform.current_step || 1;
+  currentStep = parseInt(currentStep, 10);
+  let previousStep = $("#page").attr("data-form-step") || 0;
+  previousStep = parseInt(previousStep, 10);
+
+  if (currentStep !== previousStep) {
+    const event = new CustomEvent("changeFormStep", {
+      detail: [currentStep, previousStep],
```

Comment:
I’d prefer a data object like `{current: currentStep, previous: previousStep}` as this would be unambiguous (you don’t have to remember an arbitrary order of unnamed array items).

Reply:
changed

Reply:
Thanks! Is there a particular reason to keep them nested in 'details'?

Reply:
Thanks! Is there a particular reason to keep them nested in 'detail'?

Reply:
Yes. jQuery.

Reply:
yes, it's the designated place for additional data in CustomEvent → https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/detail

Reply:
I see, that’s a good reason :wink: (wasn’t aware you have to add that manually and it's not just the name for the object)